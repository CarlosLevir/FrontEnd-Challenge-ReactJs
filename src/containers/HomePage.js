import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { getPosts } from '../services/api';
import HomeCard from '../presentations/HomeCard';

const styles = {
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center'
  }
};

class HomePage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      posts: []
    };
  }

  componentDidMount = () => {
    getPosts()
      .then((response) => {
        this.setState({
          posts: response.data
        });
      })
      .catch(error => console.log('error ', error));
  };

  renderItem = post => <HomeCard post={post} />;

  render() {
    const { posts } = this.state;
    const postsRendered = posts.map(post => this.renderItem(post));
    return <div style={styles.container}>{postsRendered}</div>;
  }
}

export default withRouter(HomePage);
