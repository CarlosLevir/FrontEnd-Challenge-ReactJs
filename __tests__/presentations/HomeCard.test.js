import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import HomePage from '../../src/presentations/HomeCard';

configure({ adapter: new Adapter() });

describe('HomePage', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HomePage />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
