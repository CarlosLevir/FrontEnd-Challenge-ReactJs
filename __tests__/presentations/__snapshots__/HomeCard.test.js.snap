// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <Uncontrolled(Panel)
      bsSize="small"
      style={
        Object {
          "height": 200,
          "margin": 5,
          "maxWidth": 400,
        }
      }
    >
      <PanelHeading
        bsClass="panel"
        componentClass="div"
      >
        test
      </PanelHeading>
      <PanelBody
        bsClass="panel"
        collapsible={false}
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
          }
        }
      >
        1
        1
        test
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          disabled={false}
        >
          Learn more
        </Button>
      </PanelBody>
    </Uncontrolled(Panel)>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Panel)
        bsSize="small"
        style={
          Object {
            "height": 200,
            "margin": 5,
            "maxWidth": 400,
          }
        }
      >
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          test
        </PanelHeading>
        <PanelBody
          bsClass="panel"
          collapsible={false}
          style={
            Object {
              "display": "flex",
              "flexWrap": "wrap",
            }
          }
        >
          1
          1
          test
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="danger"
            disabled={false}
          >
            Learn more
          </Button>
        </PanelBody>
      </Uncontrolled(Panel)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsSize": "small",
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            test
          </PanelHeading>,
          <PanelBody
            bsClass="panel"
            collapsible={false}
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          >
            1
            1
            test
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="danger"
              disabled={false}
            >
              Learn more
            </Button>
          </PanelBody>,
        ],
        "style": Object {
          "height": 200,
          "margin": 5,
          "maxWidth": 400,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": "test",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": Array [
              1,
              1,
              "test",
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="danger"
                disabled={false}
              >
                Learn more
              </Button>,
            ],
            "collapsible": false,
            "style": Object {
              "display": "flex",
              "flexWrap": "wrap",
            },
          },
          "ref": null,
          "rendered": Array [
            1,
            1,
            "test",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "danger",
                "children": "Learn more",
                "disabled": false,
              },
              "ref": null,
              "rendered": "Learn more",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Panel)
          bsSize="small"
          style={
            Object {
              "height": 200,
              "margin": 5,
              "maxWidth": 400,
            }
          }
        >
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            test
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          >
            1
            1
            test
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="danger"
              disabled={false}
            >
              Learn more
            </Button>
          </PanelBody>
        </Uncontrolled(Panel)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsSize": "small",
          "children": Array [
            <PanelHeading
              bsClass="panel"
              componentClass="div"
            >
              test
            </PanelHeading>,
            <PanelBody
              bsClass="panel"
              collapsible={false}
              style={
                Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                }
              }
            >
              1
              1
              test
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="danger"
                disabled={false}
              >
                Learn more
              </Button>
            </PanelBody>,
          ],
          "style": Object {
            "height": 200,
            "margin": 5,
            "maxWidth": 400,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": "test",
              "componentClass": "div",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": Array [
                1,
                1,
                "test",
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="danger"
                  disabled={false}
                >
                  Learn more
                </Button>,
              ],
              "collapsible": false,
              "style": Object {
                "display": "flex",
                "flexWrap": "wrap",
              },
            },
            "ref": null,
            "rendered": Array [
              1,
              1,
              "test",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "danger",
                  "children": "Learn more",
                  "disabled": false,
                },
                "ref": null,
                "rendered": "Learn more",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
